File: sys-config/Model/CMA/LOV/Nationality.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SystemConfigurator.Model.CMA.LOV;

[Table("cma_tbl_lov_nationality")]
public class Nationality
{
    [Key]
    [Column("nationality_id")]
    public Guid NationalityId { get; set; } = Guid.NewGuid();

    [Required]
    [Column("nationality_name")]
    [StringLength(80)]
    public string NationalityName { get; set; } = "";

    [Required]
    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime? UpdatedAt { get; set; }

    [Column("deleted_at")]
    public DateTime? DeletedAt { get; set; }

    // Navigation Property
    public ICollection<Customer> Customers { get; set; } = new List<Customer>();
}

File: kyc-service/src/main/java/com/devcraftlabs/kyc/model/entity/lov/Nationality.java
package com.devcraftlabs.kyc.model.entity.lov;

import java.time.Instant;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "cma_tbl_lov_nationality")
@Data
public class Nationality {

    @Id
    @Column(name = "nationality_id", nullable = false)
    private UUID nationalityId;

    @Column(name = "nationality_name", nullable = false, length = 80)
    private String nationalityName;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt = Instant.now();

    @Column(name = "updated_at")
    private Instant updatedAt;

    @Column(name = "deleted_at")
    private Instant deletedAt;
}